src/lender/tranche.sol

```diff
@@ -276,14 +278,24 @@ contract Tranche is Math, Auth, FixedPoint {
         token.burn(self, tokenAmount);
     }
 
-    function safePayout(uint currencyAmount) internal {
+    function safePayout(uint currencyAmount) internal returns(uint payoutAmount) {
         uint max = reserve.totalBalanceAvailable();
+
         if(currencyAmount > max) {
+            // currently reserve can't fulfill the entire request
             currencyAmount = max;
         }
         reserve.payout(currencyAmount);
+        return currencyAmount;
     }
 
+    function balanceCurrencyRequest() public {
```

Comment:
I think the naming now is a bit confusing. Maybe we could rename `requestCurrency` to `requestedCurrency` and rename `balanceCurrencyRequest()` to `payoutRequestedCurrency()`?